src/huggingface_hub/file_download.py

```diff
@@ -152,11 +152,7 @@ class HfFileMetadata:
     location: str
 
 
-# Do not validate `repo_id` in `hf_hub_url` for now as the `repo_id="datasets/.../..."`
-# pattern is used/advertised in Transformers examples.
-# Related: https://github.com/huggingface/huggingface_hub/pull/1029
-# TODO: set back validation in V0.12.
-# @validate_hf_hub_args
```

Comment:
I checked it and the occurrences have already been removed in `transformers` so we can start validating now instead of waiting 0.12. There has never been a proper deprecation warning anyway.